### PR TITLE
feat: add polls

### PR DESCRIPTION
**Pull request by Itrulia**
```graphql
mutation createPoll($opensAt: DateTime, $closedAt:DateTime, $question:String) {
  createPoll(opensAt: $opensAt, closedAt:$closedAt, question:$question) {
    id
    question

    answers {
      id
      answer
    }
  }
}

// vars
{
  "question": "Are you subscribed to Netflix?"
}
```

```graphql
mutation createPollExternalVoteSource($pollId:ID!, $source:String) {
  createPollExternalVoteSource(pollId: $pollId, source:$source) {
    id
    source
    voteAmounts {
      id
      amount
    }
  }
}

// vars
{
    "pollId":"cl70hbt6k0052edpxleiveqb4",
    "source":"Instagram"
}
```

```graphql
mutation voteOnPoll($answerId: ID!) {
  voteOnPoll(answerId: $answerId) {
    fingerprint
  }
}

// vars
{
  "answerId":"cl70hbt6l0053edpxx7cmkxnw"
}
```

```graphql
mutation updatePoll(
  $pollId: ID!, 
  $opensAt: DateTime, 
  $closedAt: DateTime, 
  $question: String,
  $answers: [UpdatePollAnswer!],
  $externalVoteSources: [UpdatePollExternalVoteSources!],
) {
  updatePoll(
    pollId: $pollId, 
    opensAt:$opensAt, 
    closedAt:$closedAt, 
    question:$question,
    answers:$answers,
    externalVoteSources:$externalVoteSources
  ) {
    id
    question
    opensAt
    closedAt
    answers {
      id
      answer
    }
    externalVoteSources {
      id
      source
      voteAmounts {
        id,
        amount
        answerId
      }
    }
  }
}

// vars
{
  "pollId":"cl70hbt6k0052edpxleiveqb4",
  "question":"How was your day?",
  "answers": [
    {"id":"cl70hbt6l0053edpxx7cmkxnw", "answer":"Great!"},
    {"id":"cl70hbt6l0054edpxw4x9jqhx", "answer":"Bad!"}
  ],
  "externalVoteSources": [
    {"id":"cl70hc48k0088edpxjmlc43bv", "source":"Instagram2", "voteAmounts":[
      {"id":"cl70hc48k0089edpxtk3t0di5", "amount":10},
      {"id":"cl70hc48k0090edpx2xz7bypx", "amount":30}
    ]}
  ]
}
```

### CODE REVIEW

File: packages/api/src/db/user.ts [hunk 1]
```diff
@@ -45,7 +45,9 @@ export const unselectPassword: Record<
   password: false,
   active: true,
   lastLogin: true,
-  roleIDs: true
+  roleIDs: true,
+  // Models that shouldn't be loaded by default
+  PollVote: false
```
[the remark this answers is not in the document]
Comment (Itrulia) [reply packages/api/src/db/user.ts:50]: image

on user currently all of them are like this

File: packages/api/prisma/schema.prisma [hunk 2]
```diff
@@ -686,3 +687,90 @@ model Setting {
...
+  modifiedAt DateTime @updatedAt
+
+  userId String?
+  user   User?   @relation(fields: [userId], references: [id], onDelete: Cascade)
+
+  answerId String
+  answer   PollAnswer @relation(fields: [answerId], references: [id], onDelete: Cascade)
+
+  // While the poll can be infered through the answer, having the pollId in this table
+  // allows to make sure a sure only casts 1 vote per poll
```
[the remark this answers is not in the document]
Comment (Itrulia) [reply packages/api/prisma/schema.prisma:736]: Haha yeah, thanks for noticing!

**Review by camilleory — APPROVED** [review]

**Review by camilleory — DISMISSED** [review]

**Review by michael-scheurer — COMMENTED** [review]
Outstanding work. See comments within the code. Only thing that worries me a bit is the missing sanitizing of the user input. Is this happening elsewhere?

File: packages/api/src/graphql/poll/poll.private-mutation.ts [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+  input: Pick<Prisma.PollUncheckedCreateInput, 'question' | 'opensAt' | 'closedAt'>,
+  authenticate: Context['authenticate'],
+  poll: PrismaClient['poll']
+) => {
+  const {roles} = authenticate()
+  authorise(CanCreatePoll, roles)
+
+  return poll.create({
+    data: {
+      ...input,
```
[the remark this answers is not in the document]
Comment (Itrulia) [reply packages/api/src/graphql/poll/poll.private-mutation.ts:42]: GraphQL does this

File: packages/api/src/graphql/mutation.private.ts [hunk 4]
```diff
@@ -873,6 +891,98 @@ export const GraphQLAdminMutation = new GraphQLObjectType<undefined, Context>({
...
+      type: GraphQLPollAnswer,
+      args: {
+        pollId: {type: GraphQLNonNull(GraphQLID)},
+        answer: {type: GraphQLString}
+      },
+      resolve: (
+        root,
+        {pollId, answer},
+        {authenticate, prisma: {pollExternalVoteSource, pollAnswer}}
+      ) => createPollAnswer(pollId, answer, authenticate, pollExternalVoteSource, pollAnswer)
```
[the remark this answers is not in the document]
Comment (Itrulia) [reply packages/api/src/graphql/mutation.private.ts:920]: We currently do it everywhere like this, can we keep it like this too? Else we will have incosistency.

File: packages/api/src/db/user.ts (hunk 1 above)
Comment (Itrulia) [inline packages/api/src/db/user.ts:50]: Eh this actually should go into the Exclude above

File: packages/api/src/graphql/poll/poll.ts [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+import {GraphQLDateTime} from 'graphql-iso-date'
+import {Context} from '../../context'
+import {ConnectionResult} from '../../db/common'
+import {GraphQLPageInfo} from '../common'
+import {PollSort} from './poll.private-queries'
+
+const validateVoteValue = (voteValue: unknown): number => {
+  if (typeof voteValue !== 'number') {
+    throw new GraphQLError(`Value is not a number: ${voteValue}`)
+  }
```
Comment (Itrulia) [inline packages/api/src/graphql/poll/poll.ts:32]: Missing the actual validation that it's not below 0

**Pull request merged**